stream_interfaces.go

```diff
@@ -20,56 +20,99 @@ package grpc
+// message stream and the type of the unary response message. If an error occurs
+// on the stream, it will be returned as an instance of the status package. Refer
+// to the status package documentation for more details. It is used in generated code.
 type ClientStreamingServer[Req any, Res any] interface {
+	// Recv reads a request message from the client. This method can be called
+	// repeatedly to receive the full stream of messages from the client. Recv returns
+	// (nil, io.EOF) once the end of the stream is reached.
 	Recv() (*Req, error)
+
+	// SendAndClose sends a single response message to the client and closes the stream.
+	// This method must be called once and only once after all request messages have
+	// been processed. No further methods should be called after SendAndClose.
 	SendAndClose(*Res) error
 	ServerStream
 }
 
 // BidiStreamingClient represents the client side of a bidirectional-streaming
 // (many requests, many responses) RPC. It is generic over both the type of the
-// request message stream and the type of the response message stream. It is
-// used in generated code.
+// request message stream and the type of the response message stream. If an
+// error occurs on the stream, it will be returned as an instance of the status
+// package. Refer to the status package documentation for more details. End-of-stream
+// for the client-to-server stream can be indicated by calling the CloseSend method.
+// It is used in generated code.
 type BidiStreamingClient[Req any, Res any] interface {
+	// Send sends a message to the server. This method can be called repeatedly
+	// to send messages as part of the client-to-server request stream.
 	Send(*Req) error
+
+	// Recv receives the next message from the server's response stream. This method
+	// can be called repeatedly to receive all messages sent by the server. Recv returns
+	// (nil, io.EOF) once the server-to-client stream is completely read.
 	Recv() (*Res, error)
 	ClientStream
 }
 
 // BidiStreamingServer represents the server side of a bidirectional-streaming
 // (many requests, many responses) RPC. It is generic over both the type of the
-// request message stream and the type of the response message stream. It is
+// request message stream and the type of the response message stream.  If an
+// error occurs on the stream, it will be returned as an instance of the status
+// package. Refer to the status package documentation for more details. It is
 // used in generated code.
 type BidiStreamingServer[Req any, Res any] interface {
+	// Recv receives a request message from the client. The server-side handler can
+	// repeatedly call Recv to read the request message stream. Recv returns (nil, io.EOF)
+	// once the end of the client-to-server stream is reached.
 	Recv() (*Req, error)
+
+	// Send sends a response message to the client. The server-side handler can repeatedly
+	// call Send to write to the server-to-client message stream. The end of the response
```

Comment:
`The end of the response stream is indicated by the return of the bidi method handler` should be moved up to interface dostring